src/main/java/games/strategy/engine/pbem/GenericEmailSender.java

```diff
@@ -173,10 +174,21 @@ public String getPassword() {
    */
   @Override
   public void setPassword(final String password) {
-    m_password = password;
+    m_password = passwordSaved ? password : USE_TRANSITIVE_PASSWORD;
     m_transPassword = password;
   }
 
+  @Override
+  public boolean isPasswordSaved() {
+    return passwordSaved;
+  }
+
+  @Override
+  public void setPasswordSaved(final boolean passwordSaved) {
+    this.passwordSaved = passwordSaved;
+    setPassword(m_transPassword);
```

Comment:
The purpose of this line is to ensure `m_password` will be updated appropriately if the `passwordSaved` field has changed.  (Identical code appears in `AbstractForumPoster`.)